### PR TITLE
Add initial test suite

### PR DESCRIPTION
**Pull request by cristipufu**
This PR adds an initial test suite for the Tunnelite project. The tests include:

- Basic tunnel creation test
- Tunnel start/stop functionality test

Additional tests can be added as needed to increase coverage.

### CODE REVIEW

**Review by cristipufu — COMMENTED** [review]
Thank you for adding the initial test suite to the project. Here's my review of the changes:

### Overall Structure
- ✅ Good organization with a dedicated test project
- ✅ Proper test project setup with xUnit
- ✅ Clear separation of test files

### Project Setup (Tunnelite.Tests.csproj)
- ✅ Correct target framework (net7.0)
- ✅ Appropriate package references for testing
- ✅ Good practice enabling nullable reference types
- ✅ Proper project reference to the main Tunnelite project

### Test Implementation (TunnelTests.cs)
**Positive Aspects:**
- ✅ Tests follow AAA pattern (Arrange, Act, Assert)
- ✅ Clear test names that describe the functionality being tested
- ✅ Async/await properly used in the async test

**Suggestions for Improvement:**
1. Consider adding more edge cases to increase test coverage:
   - Test tunnel creation with different configurations
   - Test error scenarios and exception handling
   - Test tunnel reconnection scenarios
   - Test concurrent operations

2. Test Method Organization:
   - Consider grouping related tests using test collections or fixtures
   - Add test categories/traits for better organization

3. Missing Validation:
   - The current tests are very basic - consider adding validation of tunnel properties
   - Add timeout handling for async operations
   - Test the actual functionality of data transfer through the tunnel

### Recommendations:
1. Add test cases for:
   - Connection failures
   - Network interruptions
   - Configuration validation
   - Performance under load
   - Resource cleanup

2. Consider adding integration tests alongside unit tests

3. Add test documentation:
   - XML comments explaining test scenarios
   - Comments for complex test arrangements

4. Consider adding test utilities:
   - Helper methods for common setup
   - Mock implementations for external dependencies
   - Test data generators

The PR provides a good foundation for testing, but would benefit from more comprehensive test coverage. I recommend expanding the test suite to cover more scenarios and edge cases.

File: tests/Tunnelite.Tests/TunnelTests.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        }
+
+        [Fact]
+        public async Task TestTunnelStartStop()
+        {
+            // Arrange
+            var tunnel = new Tunnel();
+
+            // Act & Assert
+            await tunnel.StartAsync();
```
Comment (cristipufu) [inline tests/Tunnelite.Tests/TunnelTests.cs:24]: Consider adding a timeout parameter to handle cases where the tunnel start operation takes too long. This can help catch potential connection issues early.

File: tests/Tunnelite.Tests/TunnelTests.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    public class TunnelTests
+    {
+        [Fact]
+        public void TestTunnelConnection()
+        {
+            // Arrange
+            var tunnel = new Tunnel();
+
+            // Act & Assert
+            Assert.NotNull(tunnel);
```
Comment (cristipufu) [inline tests/Tunnelite.Tests/TunnelTests.cs:14]: This test is quite basic. Consider adding validation of the tunnel's properties and state after creation. For example, check if the default configuration is properly set.

File: tests/Tunnelite.Tests/TunnelTests.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+using Xunit;
+
+namespace Tunnelite.Tests
+{
+    public class TunnelTests
```
Comment (cristipufu) [inline tests/Tunnelite.Tests/TunnelTests.cs:5]: Consider implementing IDisposable pattern for the test class to ensure proper cleanup of tunnel resources after each test. You can use the xUnit IClassFixture interface for this purpose.
